lib/ex_ami/time_server.exs

```diff
@@ -0,0 +1,144 @@
+defmodule TimeServer do
+  use GenServer
+
+  defstruct agents: %{},
+            results: %{},
+            values: [],
+            doubles: []
+
+  @name __MODULE__
+  @num_cores 2
+
+  def child_spec(args) do
+    %{
+      id: "#{@name}",
+      start: {__MODULE__, :start_link, args},
+      restart: :transient,
+      shutdown: 60,
+      type: :worker
+    }
+  end
+
+  @spec start_link(any) :: no_return
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: @name)
```

Comment:
@palexanderm I disagree. I use @name as the name of the process. Most of the time its set to __MODULE__, but the code should still work if @name is set to something else. The original code is correct since the first arg to start_link is the module. name: @name is correct since that is the process name.